features/list_categories.feature

```diff
@@ -0,0 +1,25 @@
+Feature: Visitor views list of categorized products
+  As a visitor,
+  In order to choose product more easily
+  I would like to see the products sorted in categories
+
+  Background:
+    Given the following categories exist
```

Comment:
You are using the `Category.find_or_create_by` in the table step def for creating products. So you do not really need to have this table. A bit cleaner.